test/unit_tests/apps/financial-well-being/fwb-questions.js

```diff
@@ -0,0 +1,169 @@
+                </fieldset>
+            </div>
+        </div>
+        <div class="o-form_group">
+            <input class="submit-quiz a-btn a-btn__disabled"
+                  id="submit-quiz" type="submit" value="Get my score"
+                  data-gtm-action="Questionnaire Submitted"
+                  data-gtm-label="Get my score"
+                  data-gtm-category="Financial Well-Being Tool Interaction"
+                  title="Please answer all questions to get your score">
+        </div>
+    </form>`;
+
+function triggerClickEvent( target ) {
+  var event = document.createEvent( 'Event' );
+  event.initEvent( 'click', true, true );
+  return target.dispatchEvent( event );
+}
+
+function initFwbQuestions( ) {
+  fwbQuestions = require( BASE_JS_PATH + 'apps/financial-well-being/fwb-questions' );
+  fwbQuestions.init();
+}
+
+function fillOutForm() {
+  for ( let i = 0; i <= 9; i++ ) {
+    const radioBtn = document.createElement( 'input' );
+    radioBtn.type = 'radio';
+    radioBtn.name = i + 'radio';
+    radioBtn.value = i + 'value';
+    document.body.appendChild( radioBtn );
+    triggerClickEvent( radioBtn );
+  }
+
+  const radioButtons = document.querySelectorAll( '[type="radio"]' );
+  [].forEach.call( radioButtons, function( radioElement ) {
+    triggerClickEvent( radioElement );
+  } );
+}
+
+describe( 'fwb-questions', () => {
+  jsdom();
+
+  beforeEach( () => {
+    sandbox = sinon.sandbox.create();
+    document.body.innerHTML = HTML_SNIPPET;
+    window.dataLayer = [];
+    window.tagManagerIsLoaded = true;
+    submitBtnDom = document.querySelector( '#submit-quiz' );
+    radioButtonsDom = document.querySelectorAll( '[type="radio"]' );
+  } );
+
+  afterEach( () => {
+    sandbox.restore();
+  } );
+
+  it( 'submit button should have the correct state on initialization.', () => {
+    initFwbQuestions();
+    expect( submitBtnDom.classList.contains( 'a-btn__disabled' ) )
+    .to.equal( true );
+
+    expect( submitBtnDom.title )
+    .to.equal( 'Please answer all questions to get your score' );
+  } );
+
+  it( 'submit button, if clicked, shouldnt submit the form unless all the questions are completed.',
+    () => {
+      initFwbQuestions();
+      const formSubmissionStatus = triggerClickEvent( submitBtnDom );
+      expect( formSubmissionStatus ).to.equal( false );
+    } );
+
+  it( 'submit button, if clicked, should submit the form if all the questions are completed.',
+    () => {
+      fillOutForm();
+      initFwbQuestions();
+      const formSubmissionStatus = triggerClickEvent( submitBtnDom );
+      expect( formSubmissionStatus ).to.equal( true );
+    } );
+
+  it( 'should the correct analytics when a radio button is clicked', () => {
+    initFwbQuestions();
+    triggerClickEvent( radioButtonsDom[0] );
+    expect( window.dataLayer[0] ).to.deep.equal( dataLayerEvent );
+  } );
+
+  it( 'should enable submit button if all questions completed.', () => {
+    fillOutForm();
```

Comment:
Can this be placed directly in a `beforeEach`? Group the relevant tests by a describe if needed.

Reply:
We're not always filling out the form and when we do we're not always filling it at the same time in order to test using a back button (a known error in IE)